pkg/object/mqttproxy/broker.go

```diff
@@ -450,6 +553,67 @@ func (b *Broker) sendMsgToClient(span *model.SpanContext, topic string, payload
 	}
 }
 
+func (b *Broker) processBrokerModePublish(clientID string, publish *packets.PublishPacket) {
+	if !b.spec.BrokerMode {
+		return
+	}
+	span := generateNewSpanContext(clientID, publish.TopicName)
+	subscribers, _ := b.topicMgr.findSubscribers(publish.TopicName)
+	egNames := make(map[string]struct{})
+	for clientID, subQos := range subscribers {
+		if subQos < publish.Qos {
+			continue
+		}
+		egName := b.sessionCacheMgr.getEGName(clientID)
+		if egName != b.egName {
+			egNames[egName] = struct{}{}
+			continue
+		}
+		client := b.getClient(clientID)
+		if client == nil {
+			logger.SpanDebugf(span, "client %v not on broker %v in eg %v", clientID, b.name, b.egName)
+		} else {
+			client.session.publish(span, publish.TopicName, publish.Payload, publish.Qos)
+		}
+	}
+
+	data := &HTTPJsonData{}
```

Comment:
Why broadcast? We have all the subscribers' information, we should accurately forward the packet to the subscribers' node.

Reply:
in line 563 to 578, we find subscribers eg name and put it in a map. And only send data to these egs.... It's not broadcast...

Reply:
Got. I misunderstood.